packages/amplify-cli/src/__tests__/extensions/amplify-helpers/git-manager.test.ts

```diff
@@ -32,16 +34,22 @@ const ignoreList = [
   '.secret-*',
 ];
 
-const toAppend = `${os.EOL + os.EOL + amplifyMark + os.EOL}${ignoreList.join(os.EOL)}`;
-
+const toAppend = `${os.EOL + os.EOL + amplifyMark + os.EOL}${ignoreList.join(os.EOL)}${os.EOL + amplifyEndMark + os.EOL}`;
+const legacyToAppend = `${os.EOL + os.EOL + legacyAmplifyMark + os.EOL}${ignoreList.join(os.EOL)}`;
 describe('git-manager', () => {
   beforeEach(() => {
     fsMock.writeFileSync.mockClear();
+    fsMock.readFileSync.mockClear();
+    fsMock.appendFileSync.mockClear();
+    fsMock.existsSync.mockClear();
   });
   const gitIgnoreFilePath = 'testPath';
   test('appends files to an existing .gitignore', () => {
     fsMock.existsSync.mockImplementation(() => true);
+    fsMock.readFileSync.mockImplementation(() => 'amplify/');
     insertAmplifyIgnore(gitIgnoreFilePath);
+    expect(fsMock.writeFileSync.mock.calls[0][0]).toEqual('testPath');
```

Comment:
nit: use the `gitIgnoreFilePath` variable instead of hardcoding, here and elsewhere